Docs-3317 SCALE 22.12-Beta.1 Release Notes (#1503)
truenas/documentation: content/SCALE/SCALE22.12.md

| SCALE 22.12.0 | Internal Testing Sprints | 24 November 2022 - 09 December 2022 |
| SCALE 22.12.0 | Tag | 12 December 2022 |
| SCALE 22.12.0 | Release | 13 December 2022 |
| SCALE 22.12.1 | Code-freeze | 18 January 2023 |
| SCALE 22.12.1 | Internal Testing Sprints | 19 January 2023 - 03 February 2023 |
| SCALE 22.12.1 | Tag | 06 February 2023 |
| SCALE 22.12.1 | Release | 07 February 2023 |

## Obtaining the Release

To download an <file>.iso</file> file for installing SCALE Bluefin, go to https://www.truenas.com/truenas-scale/ and click **Download**.
Manual update files are also available at this location.

To upgrade an existing SCALE install, log in to your SCALE web interface and go to **System Settings > Update**.

{{< hint warning >}}
SCALE is developed as an appliance that uses specific Linux packages with each release. Attempting to update SCALE with `apt` or methods other than the SCALE web interface can result in a nonfunctional system.
{{< /hint >}}

## 22.12-BETA.1 

**September 13, 2022**

TrueNAS SCALE 22.12-BETA.1 has been released and includes many new features and improved functionality. SCALE 22.12-Beta.1 features:

> * STIG hardening through limiting web login and API access by restricting access for non-approved IP addresses and ranges. 
>   Additional STIG hardining through disabling root login access and tying user to API ACLs (target SCALE 22.12-Beta.2).  
> * Storj iX Cloud Sync backup solution now available.
> * Apps improvments including adding Storj to the official catalog and adding a default Apps catalog exclusive for Enterprise customers (SCALE 22.12-Beta.1)
> * Redesign of Storage web UI inclduing new dashboards for Storage, Pools, Dashboards, Devices and other storage related areas
> * Enclosure management for all iXsystems platforms 
> * Improved clustering over the Angelfish clustered SMB (aka Windows storage).

Reviewer: Requested change from Kris
```suggestion
* Redesign of Storage web UI including new dashboards for Storage, Pools, Dashboards, Devices and other storage related areas
* Storj iX Cloud Sync backup solution now available.
* Apps improvements including adding Storj to the official catalog and adding a default Apps catalog exclusive for Enterprise customers (SCALE 22.12-Beta.1)
* STIG hardening through limiting web login and API access by restricting access for non-approved IP addresses and ranges. 
  Additional STIG hardening through disabling root login access and tying user to API ACLs (target SCALE 22.12-Beta.2).  
* Enclosure management for all iXsystems platforms 
* Improved clustering over the Angelfish clustered SMB (aka Windows storage).
```

* 8GB of RAM (More is better)
* 20GB Boot Device

## Software Lifecycle

{{< include file="/static/includes/General/LifecycleTable.html.part" html="true" >}}

{{< include file="/content/_includes/SoftwareStatusPage.md" type="page" >}}

## SCALE Schedule

{{< include file="/content/_includes/ReleaseScheduleWarning.md" type="page" >}}

| Version | Checkpoint | Scheduled Date |
|---------|------------|----------------|

> | SCALE 22.12.BETA.1 | Code-freeze |  |
> | SCALE 22.12.BETA.1 | Internal Testing Sprints |  |
> | SCALE 22.12.BETA.1 | Tag |  |
> | SCALE 22.12.BETA.1 | Release |  |
> | SCALE 22.12.BETA.2 | Code-freeze |  |
> | SCALE 22.12.BETA.2 | Internal Testing Sprints |  |
> | SCALE 22.12.BETA.2 | Tag |  |
> | SCALE 22.12.BETA.2 | Release |  |
> | SCALE 22.12.RC.1 | Code-freeze |  |
> | SCALE 22.12.RC.1 | Internal Testing Sprints |  |
> | SCALE 22.12.RC.1 | Tag |  |
> | SCALE 22.12.RC.1 | Release |  |
> | SCALE 22.12.0 | Code-freeze |  |
> | SCALE 22.12.0 | Internal Testing Sprints |  |
> | SCALE 22.12.0 | Tag |  |
> | SCALE 22.12.0 | Release |  |
> | SCALE 22.12.1 | Code-freeze |  |
> | SCALE 22.12.1 | Internal Testing Sprints |  |
> | SCALE 22.12.1 | Tag |  |
> | SCALE 22.12.1 | Release |  |

Reviewer: Adding anticipated Bluefin release scheduling.
```suggestion
| SCALE 22.12.BETA.1 | Code-freeze | 24 August 2022 |
| SCALE 22.12.BETA.1 | Internal Testing Sprints | 29 August 2022 - 09 September 2022 |
| SCALE 22.12.BETA.1 | Tag | 12 September 2022 |
| SCALE 22.12.BETA.1 | Release | 13 September 2022 |
| SCALE 22.12.BETA.2 | Code-freeze | 28 September 2022 |
| SCALE 22.12.BETA.2 | Internal Testing Sprints | 03 October 2022 - 14 October 2022 |
| SCALE 22.12.BETA.2 | Tag | 17 October 2022 |
| SCALE 22.12.BETA.2 | Release | 18 October 2022 |
| SCALE 22.12.RC.1 | Code-freeze | 26 October 2022 |
| SCALE 22.12.RC.1 | Internal Testing Sprints | 31 October 2022 - 11 November 2022 |
| SCALE 22.12.RC.1 | Tag | 14 November 2022 |
| SCALE 22.12.RC.1 | Release | 15 November 2022 |
| SCALE 22.12.0 | Code-freeze | 23 November 2022 |
| SCALE 22.12.0 | Internal Testing Sprints | 24 November 2022 - 09 December 2022 |
| SCALE 22.12.0 | Tag | 12 December 2022 |
| SCALE 22.12.0 | Release | 13 December 2022 |
| SCALE 22.12.1 | Code-freeze | 18 January 2023 |
| SCALE 22.12.1 | Internal Testing Sprints | 19 January 2023 - 03 February 2023 |
| SCALE 22.12.1 | Tag | 06 February 2023 |
| SCALE 22.12.1 | Release | 07 February 2023 |
```

[NAS-116411](https://ixsystems.atlassian.net/browse/NAS-116411) Information for different roles attached to a dataset or child datasets
[NAS-116410](https://ixsystems.atlassian.net/browse/NAS-116410) The Roles details Card/Widget
[NAS-116406](https://ixsystems.atlassian.net/browse/NAS-116406) Data Protection Card/Widget
[NAS-116403](https://ixsystems.atlassian.net/browse/NAS-116403) Dataset Management Page
[NAS-116397](https://ixsystems.atlassian.net/browse/NAS-116397) Disk Health card
[NAS-116393](https://ixsystems.atlassian.net/browse/NAS-116393) Pool Topology details widget/card
[NAS-116391](https://ixsystems.atlassian.net/browse/NAS-116391) Single Pool details component on Pools dashboard
[NAS-116389](https://ixsystems.atlassian.net/browse/NAS-116389) Create Pools Dashboard
[NAS-114198](https://ixsystems.atlassian.net/browse/NAS-114198) Advanced boot options for SCALE \(udev rules and kernel parameters\)
[NAS-111020](https://ixsystems.atlassian.net/browse/NAS-111020) Assigning Host USB device to a Guest VM in SCALE
[NAS-102765](https://ixsystems.atlassian.net/browse/NAS-102765) Ask for EC2 Instance ID when setting initial root password## Known Issues

> | Seen In | Key | Summary | Workaround | Resolved In |
> |---------|-----|---------|------------|-------------|
> | 22.02.3 | <a href="https://ixsystems.atlassian.net/browse/NAS-117581" target="_blank">NAS-117581</a> | Launch Docker Image button is disabled. | On the <b>Apps</b> page, select the <b>Available Applications</b> tab before trying to click the button. | 22.02.3.1 |
> | 22.02.1 |<a href="https://ixsystems.atlassian.net/browse/NAS-116473" target="_blank">NAS-116473</a> | Large Drive Count Issues | iX is investigating isuses with booting SCALE on systems with more than 100 Disks. | 22.12-RC.1 |
> | 22.02.0 | <a href="https://jira.ixsystems.com/browse/NAS-115238" target="_blank">NAS-115238</a> | Removed drive from pool does not degrade pool status (SCALE). | Issue is being investigated and a fix provided in a future release | Targeted 22.02.4 |
> | 22.02.0-RC.2 |  | Cosmetic issue with update trains when updating from SCALE 22.02.0-RC.2. | After updating from 22.02.0-RC.2, the previous update train might show in <b>System Settings > Update</b>. This is a cosmetic issue only and can be ignored. |  |
> |  |  | Unable to mount an NFS export after migrating from CORE > SCALE or updating to 22.02.0. | The <file>/etc/exports</file> file is no longer generated when the NFS configuration contains <i>mapall</i> or <i>maproot</i> entries for unknown users or groups. This can impact users who previously had a mapping group set to <i>wheel</i>, which does not exist in SCALE. If you are unable to mount an NFS export, review your NFS share configuration and change any <i>wheel</i> entries to something specific for your environment or <i>root</i>. |  |
> |  |  | ZFS feature flag has been removed. | See <a href="#zfs-feature-flag-removal">ZFS Feature Flag Removal</a> for more information. |  |
> |  |  | SCALE Gluster/Cluster. | Gluster/Cluster features are still in testing.  Administrators should use caution when deploying and avoid use with critical data. |  |
> |  | <a href="https://jira.ixsystems.com/browse/NAS-110263" target="_blank">NAS-110263</a> | AFP sharing is removed from TrueNAS SCALE. The protocol is deprecated and no longer receives development effort or security fixes. | TrueNAS SCALE automatically migrates any existing AFP shares into an SMB configuration that is preset to function like an AFP share. | 21.06-BETA.1 |
> | 21.06-BETA.1 | <a href="https://jira.ixsystems.com/browse/NAS-111547" target="_blank">NAS-111547</a> | ZFS shouldn't count vdev IO errors on hotplug removal | Pool status isn't being updated immediately on disk exchange events. | Targeted 22.12 |

Reviewer: Duplicate text
```suggestion

```

## Known Issues

| Seen In | Key | Summary | Workaround | Resolved In |
|---------|-----|---------|------------|-------------|

> | 22.02.3 | <a href="https://ixsystems.atlassian.net/browse/NAS-117581" target="_blank">NAS-117581</a> | Launch Docker Image button is disabled. | On the <b>Apps</b> page, select the <b>Available Applications</b> tab before trying to click the button. | 22.02.3.1 |
> | 22.02.1 |<a href="https://ixsystems.atlassian.net/browse/NAS-116473" target="_blank">NAS-116473</a> | Large Drive Count Issues | iX is investigating isuses with booting SCALE on systems with more than 100 Disks. | 22.12-RC.1 |
> | 22.02.0 | <a href="https://jira.ixsystems.com/browse/NAS-115238" target="_blank">NAS-115238</a> | Removed drive from pool does not degrade pool status (SCALE). | Issue is being investigated and a fix provided in a future release | Targeted 22.02.4 |
> | 22.02.0-RC.2 |  | Cosmetic issue with update trains when updating from SCALE 22.02.0-RC.2. | After updating from 22.02.0-RC.2, the previous update train might show in <b>System Settings > Update</b>. This is a cosmetic issue only and can be ignored. |  |

Reviewer: Can be removed due to Angelfish only.
```suggestion
| 22.02.3 | <a href="https://ixsystems.atlassian.net/browse/NAS-117581" target="_blank">NAS-117581</a> | Launch Docker Image button is disabled. | On the <b>Apps</b> page, select the <b>Available Applications</b> tab before trying to click the button. | 22.02.3.1 |
| 22.02.1 |<a href="https://ixsystems.atlassian.net/browse/NAS-116473" target="_blank">NAS-116473</a> | Large Drive Count Issues | iX is investigating isuses with booting SCALE on systems with more than 100 Disks. | 22.12-RC.1 |
| 22.02.0 | <a href="https://jira.ixsystems.com/browse/NAS-115238" target="_blank">NAS-115238</a> | Removed drive from pool does not degrade pool status (SCALE). | Issue is being investigated and a fix provided in a future release | Targeted 22.02.4 |
| 22.02.0-RC.2 |  | Cosmetic issue with update trains when updating from SCALE 22.02.0-RC.2. | After updating from 22.02.0-RC.2, the previous update train might show in <b>System Settings > Update</b>. This is a cosmetic issue only and can be ignored. |  |
```

Author: Excellent suggested changes!

|  |  | Unable to mount an NFS export after migrating from CORE > SCALE or updating to 22.02.0. | The <file>/etc/exports</file> file is no longer generated when the NFS configuration contains <i>mapall</i> or <i>maproot</i> entries for unknown users or groups. This can impact users who previously had a mapping group set to <i>wheel</i>, which does not exist in SCALE. If you are unable to mount an NFS export, review your NFS share configuration and change any <i>wheel</i> entries to something specific for your environment or <i>root</i>. |  |

> |  |  | ZFS feature flag has been removed. | See <a href="#zfs-feature-flag-removal">ZFS Feature Flag Removal</a> for more information. |  |

Reviewer: Remove due to Angelfish only.
```suggestion

```

| Feature Flag | GUID | Dependencies | Description |
|--------------|------|--------------|-------------|
| blake3 | org.openzfs.blake3 | extensible)dataset | Enables use of the BLAKE3 hash algorithmfor checksum and dedup. BLAKE3 is a secure hash algorithm focused on high performance. When enabled, the adminstrator can turn on the blake3 checksum on any dataset using `zfs set checksum=blake dset` [see zfs-set(8)](https://openzfs.github.io/openzfs-docs/man/8/zfs-set.8.html). |
| head_errlog | com.delphix:head_errlog | n/a | Enables the upgraded version of `errlog`. The error log of each head dataset is stored separately in the zap object and keyed by the head id. Every dataset affected by an error block is listed in the output of `zpool status`. |
| zilsaxattr | org.openzfs:zilsaxattr | extensible_dataset | Enables `xattr-sa` extended attribute logging in the ZIL. If enabled, extended attribute changes from both `xattrdir=dir` and `xattr=sa` are guaranteed to be durable if either `sync=always` is set for the dataset when a change is made or sync(2) is called on the dataset after making changes. |

> ## ZFS Feature Flag Removal 
> {{< expand "Expand if you created or upgraded a pool using a TrueNAS SCALE nightly build dated between June 29, 2021 and July 15, 2021" "v" >}}
> ### Executive Summary
> 
> * ZFS `xattr_compat` feature flag removed
> 
> ### How to tell if I'm impacted by this change
> 
> * Users who created or upgraded a pool using a TrueNAS SCALE nightly build dated between June 29, 2021 and July 15, 2021 are impacted by this change.
> * Users who have manually set `xattr_compat=all` on a dataset and written an xattr are impacted by this change.
> * If unsure, you can verify a pool's status of the `xattr_compat` feature flag. If the flag is in the active state, you are impacted by this change.
>  
> ![ZFSFeatureFlagRemovalExample](/images/SCALE/ZFSFeatureFlagsRemovalExample5.png "Feature Flag Status") 
>  
> ### How to resolve this if I am impacted
> 
> * Any pool that has had the feature active, must be backed up and restored into a pool created on a version of ZFS without the feature. For details on how to perform data protection procedures, please refer to the TrueNAS SCALE [Data Protection]({{< relref "/content/SCALE/SCALEUIReference/DataProtection/_index.md" >}}) documentation.
> 
> ### Technical details behind the change
> 
> See the [ZFS Feature Flags Removal article]({{< relref "SCALEZFSFlagRemoval.md" >}}) for more information.
> {{< /expand >}}

Reviewer: I believe this can be removed as it does not apply to Bluefin nightly builds (applied to Angelfish nightlies only)
```suggestion

```
